### PR TITLE
test: fix flake with TestNamespaceRepo

### PR DESCRIPTION
**Pull request by sdowell**
The test used a short timeout to wait for the Deployment to become current, which can be flaky on autopilot. The test does not really care whether the Deployment reconciles, so the assertion is changed to a simple existence check. Additionally, if the test fails in the middle it may not have permission to manage/delete the Deployment. A cleanup block is added to ensure the Deployment can be deleted in cleanup.

### CODE REVIEW

File: e2e/testcases/namespace_repo_test.go [hunk 1]
```diff
@@ -147,9 +151,8 @@ func validateRepoSyncRBAC(nt *nomostest.NT, ns string, nsRepo *gitproviders.Repo
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
-	if err := nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), "hello-world", ns,
-		testwatcher.WatchTimeout(30*time.Second)); err != nil {
-		nt.T.Fatalf("deployment hello-word not current: %v", err)
+	if err := nt.Validate("hello-world", ns, &appsv1.Deployment{}); err != nil {
+		nt.T.Fatalf("deployment hello-word not found: %v", err)
```
Comment (tiffanny29631) [inline e2e/testcases/namespace_repo_test.go:155]: nit:world

**Review by tiffanny29631 — COMMENTED** [review]
/lgtm

File: e2e/testcases/namespace_repo_test.go (hunk 1 above)
Comment (sdowell) [reply e2e/testcases/namespace_repo_test.go]: fixed

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**